resources/indexer/indexer.py

```diff
@@ -670,14 +635,140 @@ def gather_file_info(dirs, site, basic_only=False):
+            return True
+    return False
+
+
+def process_work(paths, args, blacklist):
+    """Wrap async function, process_paths. Return files nested under any directories."""
+    if not isinstance(paths, list):
+        raise TypeError(f'`paths` object is not list {paths}')
+    if not paths:
+        return []
+
+    # Check blacklist
+    paths = [p for p in paths if not path_in_blacklist(p, blacklist)]
+
+    # Process Paths
+    fc_rc = RestClient(args.url, token=args.token, timeout=args.timeout, retries=args.retries)
+    manager = MetadataManager(args.site, args.basic_only)
+    sub_files = asyncio.get_event_loop().run_until_complete(
+        process_paths(paths, manager, fc_rc, args.no_patch))
+
+    fc_rc.close()
+    return sub_files
+
+
+def check_path(path):
+    """Check if path is rooted at a white-listed root path."""
+    for root in ACCEPTED_ROOTS:
+        if path.startswith(root):
+            return
+    message = f"{path} is not rooted at: {', '.join(ACCEPTED_ROOTS)}"
+    logging.critical(message)
+    raise Exception(f'Invalid path ({message}).')
+
+
+def gather_file_info(starting_paths, blacklist, args):
+    """Gather and post metadata from files rooted at starting_paths. Do this multi-processed."""
+    # Get full paths
+    starting_paths = [os.path.abspath(p) for p in starting_paths]
+    for p in starting_paths:
+        check_path(p)
+
+    # Traverse paths and process files
+    futures = []
+    with ProcessPoolExecutor() as pool:
+        queue = starting_paths
+        split = math.ceil(len(queue) / args.processes)
+        while futures or queue:
+            logging.debug(f'Queue: {len(queue)}.')
+            # Divvy up queue among available worker(s). Each worker gets 1/nth of the queue.
+            if queue:
+                queue.sort()
+                while args.processes != len(futures):
+                    paths, queue = queue[:split], queue[split:]
+                    logging.debug(f'Worker Assigned: {len(futures)+1}/{args.processes} ({len(paths)} paths).')
+                    futures.append(pool.submit(process_work, paths, args, blacklist))
+            logging.debug(f'Workers: {len(futures)} {futures}.')
+            # Extend the queue
+            while not futures[0].done():  # concurrent.futures.wait(FIRST_COMPLETED) is slower
+                sleep(0.1)
+            future = futures.pop(0)
+            result = future.result()
+            if result:
+                queue.extend(result)
+                split = math.ceil(len(queue) / args.processes)
+            logging.debug(f'Worker finished: {future} (enqueued {len(result)}).')
+
+
+def sorted_unique(infile, others=None):
+    """Read in lines from file, aggregate with those in others list. Return all unique lines, sorted."""
+    lines = []
+    if others:
+        lines.extend(others)
+    if infile:
+        with open(infile) as file:
+            lines.extend([l.rstrip() for l in file])
```

Comment:
```suggestion
            lines.extend(l.rstrip() for l in file)
```
You shouldn't need to make the list, and can pass the generator to `extend`.